mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,297 @@
+          then params in the the wrapped estimator will also be logged, the nested param key
+          will be `{estimator_uid}.{param_name}`
+
+      **Tags**
+        - An estimator class name (e.g. "LinearRegression").
+        - A fully qualified estimator class name
+          (e.g. "pyspark.ml.regression.LinearRegression").
+
+      **Artifacts**
+        - An MLflow Model with the :py:mod:`mlflow.spark` flavor containing a fitted estimator
+          (logged by :py:func:`mlflow.spark.log_model()`). Note that large models may not be
+          autologged for performance and storage space considerations, and autologging for
+          Pipelines and hyperparameter tuning meta-estimators (e.g. CrossValidator) is not yet
+          supported.
+          See ``log_models`` param below for details.
+
+
+    :param log_models: If ``True``, if trained models are in allowlist, they are logged as MLflow
+                       model artifacts. If ``False``, trained models are not logged.
+                       Note: the built-in allowlist excludes some models (e.g. ALS models) which
+                       can be large. To specify a custom allowlist, create a file containing a
+                       newline-delimited list of fully-qualified estimator classnames, and set
+                       the "spark.mlflow.pysparkml.autolog.logModelAllowlistFile" Spark config
+                       to the path of your allowlist file.
+    :param disable: If ``True``, disables the scikit-learn autologging integration. If ``False``,
+                    enables the pyspark ML autologging integration.
+    :param exclusive: If ``True``, autologged content is not logged to user-created fluent runs.
+                      If ``False``, autologged content is logged to the active fluent run,
+                      which may be user-created.
+    :param disable_for_unsupported_versions: If ``True``, disable autologging for versions of
+                      pyspark that have not been tested against this version of the MLflow
+                      client or are incompatible.
+    :param silent: If ``True``, suppress all event logs and warnings from MLflow during pyspark ML
+                   autologging. If ``False``, show all events and warnings during pyspark ML
+                   autologging.
+
+    **The default log model allowlist in mlflow**
+        .. literalinclude:: ../../../mlflow/pyspark/ml/log_model_allowlist.txt
+           :language: text
+    """
+    from mlflow.utils.validation import (
+        MAX_PARAMS_TAGS_PER_BATCH,
+        MAX_PARAM_VAL_LENGTH,
+        MAX_ENTITY_KEY_LENGTH,
+    )
+    from pyspark.ml.base import Estimator
+    from mlflow.utils._spark_utils import _get_active_spark_session
+
+    global _log_model_allowlist
+
+    spark_session = _get_active_spark_session()
+    if spark_session:
+        _log_model_allowlist = _read_log_model_allowlist(spark_session)
```

Comment:
Looks like `_log_model_allowlist` will remain `None` if a spark session doesn't exist.

Reply:
I think we could do something like this:

```python
def _get_allowlist_path():
    spark_session = _get_active_spark_session()
    default_allowlist_path = get_resource(...)

    if spark_session:
        return spark_session.sparkContext._conf.get(
            "spark.mlflow.pysparkml.autolog.logModelAllowlistFile", default_allowlist_path
        )
    return default_allowlist_path

allowlist_path = _get_allowlist_path()
allowlist = _read_log_model_allowlist_from_file(allowlist_path)
```
```

Reply:
We'd better add exception handling while reading custom allowlist file. I update code.